Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,377 @@
+                foreach (var dllFile in dllFiles)
+                {
+                    ProcessDll(dllFile, batchPath);
+                }
+
+                if (Config == "Debug")
+                {
+                    UpdateAddInRegistration();
+                }
+            }
+            catch (Exception ex)
+            {
+                this.LogError(ex);
+                result = false;
+            }
+
+            return result;
+        }
+
+        private string FormatParameterList()
+        {
+            var sb = new StringBuilder();
+            sb.AppendLine("Parameters provided:")
+              .AppendLine($"          Config: {Config}")
+              .AppendLine($"     NetToolsDir: {NetToolsDir}")
+              .AppendLine($"     WixToolsDir: {WixToolsDir}")
+              .AppendLine($"       SourceDir: {SourceDir}")
+              .AppendLine($"       TargetDir: {TargetDir}")
+              .AppendLine($"      ProjectDir: {ProjectDir}")
+              .AppendLine($"     IncludesDir: {IncludeDir}")
+              .AppendLine($"  FilesToExtract: {FilesToExtract}")
+              .AppendLine(string.Empty);
+
+            return sb.ToString();
+        }
+
+        private bool TryGetVCToolsPath(out string batchPath)
+        {
+            var configuration = new SetupConfiguration();
+            var enumInstances = configuration.EnumInstances();
+            int fetched;
+            var instances = new ISetupInstance[1];
+            do
+            {
+                enumInstances.Next(1, instances, out fetched);
+
+                if (fetched <= 0)
+                {
+                    continue;
+                }
+
+                if (!(instances[0] is ISetupInstance2 instance))
+                {
+                    continue;
+                }
+
+                var packages = instance.GetPackages();
+                foreach (var package in packages)
+                {
+                    if (package.GetId() != "Microsoft.VisualStudio.Component.VC.Tools.x86.x64")
+                    {
+                        continue;
+                    }
+
+                    var rootPath = instance.ResolvePath();
+                    batchPath = Directory.GetFiles(rootPath, "VsDevCmd.bat", SearchOption.AllDirectories)
+                        .FirstOrDefault();
+
+                    return true;
+                }
+            } while (fetched > 0);
+
+            batchPath = null;
+            return false;
+        }
+
+        private void CleanOldImports(string dir)
+        {
+            this.LogMessage("Cleaing out old imports");
+
+            var files = Directory.GetFiles(dir, "DebugRegistryEntries.reg.imported_*.txt").OrderByDescending(f => f);
+            var i = 0;
+            foreach (var file in files)
+            {
+                if (i > MaxNumberOfPreviousRegistrationFiles)
+                {
+                    this.LogMessage($"Deleting {file}");
+                    File.Delete(file);
```

Comment:
Do we care if this throws?

Reply:
Actually, no, we shouldn't. I'll make the task log & swallow it.